applications/nrf5340_audio/src/modules/display/button_tab.c

```diff
@@ -0,0 +1,312 @@
+	lv_obj_set_style_radius(button, LV_RADIUS_CIRCLE, 0);
+	lv_obj_align(button, button_position, 0, 0);
+	label = lv_label_create(button);
+	lv_label_set_text(label, label_text);
+	lv_obj_set_style_text_font(label, font_size, LV_STATE_DEFAULT);
+	lv_obj_align(label, LV_ALIGN_CENTER, 0, 0);
+	lv_obj_add_event_cb(button, button_callback_function, LV_EVENT_CLICKED, NULL);
+}
+static void play_pause_button_create(lv_obj_t *current_screen, char *label_text,
+				     lv_align_t button_position,
+				     void (*button_callback_function)(lv_event_t *e),
+				     const lv_font_t *font_size)
+{
+	lv_obj_t *play_pause_button = lv_btn_create(current_screen);
+
+	lv_obj_set_size(play_pause_button, 45, 45);
+	lv_obj_set_style_radius(play_pause_button, LV_RADIUS_CIRCLE, 0);
+	lv_obj_align(play_pause_button, button_position, 0, 0);
+	play_pause_label = lv_label_create(play_pause_button);
+	lv_label_set_text(play_pause_label, label_text);
+	lv_obj_set_style_text_font(play_pause_label, font_size, LV_STATE_DEFAULT);
+	lv_obj_align(play_pause_label, LV_ALIGN_CENTER, 0, 0);
+	lv_obj_add_event_cb(play_pause_button, button_callback_function, LV_EVENT_CLICKED, NULL);
+}
+
+static void play_pause_button_event_cb(lv_event_t *e)
+{
+	int ret;
+	struct button_msg msg;
+
+	msg.button_action = BUTTON_PRESS;
+	msg.button_pin = BUTTON_PLAY_PAUSE;
+
+	ret = zbus_chan_pub(&button_chan, &msg, K_NO_WAIT);
+	if (ret) {
+		LOG_ERR("Failed to publish button msg, ret: %d", ret);
+	}
+}
+
+static void volume_up_button_event_cb(lv_event_t *e)
+{
+	int ret;
+	struct button_msg msg;
+
+	msg.button_action = BUTTON_PRESS;
+	msg.button_pin = BUTTON_VOLUME_UP;
+
+	ret = zbus_chan_pub(&button_chan, &msg, K_NO_WAIT);
+	if (ret) {
+		LOG_ERR("Failed to publish button msg, ret: %d", ret);
+	}
+}
+
+static void volume_down_button_event_cb(lv_event_t *e)
+
+{
+	int ret;
+	struct button_msg msg;
+
+	msg.button_action = BUTTON_PRESS;
+	msg.button_pin = BUTTON_VOLUME_DOWN;
+
+	ret = zbus_chan_pub(&button_chan, &msg, K_NO_WAIT);
+	if (ret) {
+		LOG_ERR("Failed to publish button msg, ret: %d", ret);
+	}
+}
+
+static void btn4_button_event_cb(lv_event_t *e)
+{
+	int ret;
+	struct button_msg msg;
+
+	msg.button_action = BUTTON_PRESS;
+	msg.button_pin = BUTTON_4;
+
+	ret = zbus_chan_pub(&button_chan, &msg, K_NO_WAIT);
+	if (ret) {
+		LOG_ERR("Failed to publish button msg, ret: %d", ret);
+	}
+}
+
+static void btn5_button_event_cb(lv_event_t *e)
+{
+	int ret;
+	struct button_msg msg;
+
+	msg.button_action = BUTTON_PRESS;
+	msg.button_pin = BUTTON_5;
+
+	ret = zbus_chan_pub(&button_chan, &msg, K_NO_WAIT);
+	if (ret) {
+		LOG_ERR("Failed to publish button msg, ret: %d", ret);
+	}
+}
+
+static void vu_bar_update(int vu_val)
+{
+	lv_bar_set_value(vu_bar, vu_val, LV_ANIM_OFF);
+}
+static void streaming_state_update(lv_obj_t *play_pause_button)
+{
+	int ret;
+	const struct zbus_channel *chan;
+
+	ret = zbus_sub_wait(&le_audio_evt_sub_display, &chan, K_NO_WAIT);
+	if (ret == 0) {
```

Comment:
The sub wait returns 0 when there is a new message. If the switch case isnt wrapped like this it will update the screen more often than needed. The last if is redundant thought, and is now removed